### PR TITLE
Behat - Group tests by content type and use Scenario outline

### PR DESCRIPTION
**Pull request by alzz**
Group behat context by cotent type.
Use scenario outline functionality because it can be expanded more easily without repeating code.
Example creating content type test for multiple roles.

### CODE REVIEW

**Comment by omarlopesino** [comment]
Please rename 'content types' folder by just `content`.

File: tests/behat/features/content_types/article_create.feature [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    And I should see "Title field is required."
+
+    Examples:
+      | role           |
+      | administrator  |
+
+
+  @error @api @article @create @anonymous
+  Scenario: Simple Article creation anonymous access
+    When I go to "/node/add/article"
```
Comment (omarlopesino) [inline tests/behat/features/content_types/article_create.feature]:
```suggestion
    Given I am an anonymous user
    When I go to "/node/add/article"
```

File: tests/behat/features/content_types/article_delete.feature [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    Then I should see the message "The Article Behat article test has been deleted"
+
+    Examples:
+      | role           |
+      | administrator  |
+
+
+  @error @api @article @delete @anonymous
+  Scenario: Simple Article delete anonymous access
+    When I go to "delete" of the "node" entity with label "Behat article test"
```
Comment (omarlopesino) [inline tests/behat/features/content_types/article_delete.feature]:
```suggestion
    Given I am an anonymous user
    When I go to "delete" of the "node" entity with label "Behat article test"
```

File: tests/behat/features/content_types/article_edit.feature [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    Then I should see "Behat article test edited"
+
+    Examples:
+      | role           |
+      | administrator  |
+
+
+  @error @api @article @edit @anonymous
+  Scenario: Simple Article edit anonymous access
+    When I go to "edit" of the "node" entity with label "Behat article test"
```
Comment (mistermoper) [inline tests/behat/features/content_types/article_edit.feature:26]:
```suggestion
    Given I am an anonymous user
    When I go to "edit" of the "node" entity with label "Behat article test"
```

File: tests/behat/features/content_types/article_edit.feature [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    Then the response status code should be 403
+
+  @error @api @article @edit @authenticated
+  Scenario Outline: Simple Article edit authenticated access
+    When I go to "edit" of the "node" entity with label "Behat article test"
+    Then the response status code should be 403
+
+    Examples:
+      | role           |
```
Comment (mistermoper) [inline tests/behat/features/content_types/article_edit.feature:35]:
```suggestion
      | role             |
      | anonymous |
```

File: tests/behat/features/content_types/article_view.feature [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+    Examples:
+      | role           |
+      | administrator  |
+      | authenticated  |
+
+
+  @api @article @view @anonymous
+  Scenario: Simple Article view anonymous access
+    When I go to the "node" entity with label "Behat article test"
```
Comment (omarlopesino) [inline tests/behat/features/content_types/article_view.feature]:
```suggestion
    Given I am an anonymous user
    When I go to the "node" entity with label "Behat article test"
```

File: tests/behat/features/content_types/page_create.feature [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    When I go to "/node/add/page"
+    Then the response status code should be 403
+
+  @error @api @page @create @authenticated
+  Scenario Outline: Simple Page creation authenticated access
+    When I go to "/node/add/page"
+    Then the response status code should be 403
+
+    Examples:
+      | role           |
```
Comment (omarlopesino) [inline tests/behat/features/content_types/page_create.feature]:
```suggestion
      | role             |
      | anonymous |
```

File: tests/behat/features/content_types/page_create.feature [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    When I go to "/node/add/page"
+    And I press "Save"
+    And I should see "Title field is required."
+
+    Examples:
+      | role           |
+      | administrator  |
+
+
+  @error @api @page @create @anonymous
```
Comment (omarlopesino) [inline tests/behat/features/content_types/page_create.feature]: This test is not necessary, the next test that does the outline steps is enough.

File: tests/behat/features/content_types/article_edit.feature [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+    And I press the "Save" button
+    Then I should see "Behat article test edited"
+
+    Examples:
+      | role           |
+      | administrator  |
+
+
+  @error @api @article @edit @anonymous
+  Scenario: Simple Article edit anonymous access
```
Comment (mistermoper) [inline tests/behat/features/content_types/article_edit.feature:25]: This test is not necessary, the next test that does the outline steps is enough.

File: tests/behat/features/content_types/page_delete.feature [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+    Then I should see the message "The Basic page Behat page test has been deleted"
+
+    Examples:
+      | role           |
+      | administrator  |
+
+
+  @error @api @page @delete @anonymous
+  Scenario: Simple Page delete anonymous access
+    When I go to "delete" of the "node" entity with label "Behat page test"
```
Comment (mistermoper) [inline tests/behat/features/content_types/page_delete.feature:27]: This test is not necessary, the next test that does the outline steps is enough.

File: tests/behat/features/content_types/page_edit.feature [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+    Then I should see "Behat page test edited"
+
+    Examples:
+      | role           |
+      | administrator  |
+
+
+  @error @api @page @edit @anonymous
+  Scenario: Simple Page edit anonymous access
+    When I go to "edit" of the "node" entity with label "Behat page test"
```
Comment (mistermoper) [inline tests/behat/features/content_types/page_edit.feature:26]: This test is not necessary, the next test that does the outline steps is enough.

File: tests/behat/features/content_types/page_view.feature [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+
+    Examples:
+      | role           |
+      | administrator  |
+      | authenticated  |
+
+
+  @api @page @view @anonymous
+  Scenario: Simple Page view anonymous access
+    When I go to the "node" entity with label "Behat page test"
```
Comment (omarlopesino) [inline tests/behat/features/content_types/page_view.feature]:
```suggestion
    Given I am an anonymous user
    When I go to the "node" entity with label "Behat page test"
```

**Review by omarlopesino — CHANGES_REQUESTED** [review]

File: tests/behat/features/content_types/page_create.feature (hunk 6 above)
Comment (alzz) [reply tests/behat/features/content_types/page_create.feature]: It is possible to add the anonymous role in the role table?
In that case I can simplify even more some scenarios

Comment (omarlopesino) [reply tests/behat/features/content_types/page_create.feature]: No. I didn't notice that. Then I'll resolve all points related to this.

File: tests/behat/features/content_types/page_delete.feature [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+    Then the response status code should be 403
+
+  @error @api @page @delete @authenticated
+  Scenario Outline: Simple Page delete authenticated access
+    When I go to "delete" of the "node" entity with label "Behat page test"
```
Comment (omarlopesino) [inline tests/behat/features/content_types/page_delete.feature]:
```suggestion
    Given I am logged in as a user with the <role> role
    When I go to "delete" of the "node" entity with label "Behat page test"
```

File: tests/behat/features/content_types/page_create.feature [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+  Scenario: Simple Page creation anonymous access
+    When I go to "/node/add/page"
+    Then the response status code should be 403
+
+  @error @api @page @create @authenticated
+  Scenario Outline: Simple Page creation authenticated access
+    When I go to "/node/add/page"
```
Comment (omarlopesino) [inline tests/behat/features/content_types/page_create.feature]:
```suggestion
    Given I am logged in as a user with the <role> role
    When I go to "/node/add/page"
```

File: tests/behat/features/content_types/page_create.feature [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+    And I should see "Title field is required."
+
+    Examples:
+      | role           |
+      | administrator  |
+
+
+  @error @api @page @create @anonymous
+  Scenario: Simple Page creation anonymous access
+    When I go to "/node/add/page"
```
Comment (omarlopesino) [inline tests/behat/features/content_types/page_create.feature]:
```suggestion
    Given I am an anonymous user
    When I go to "/node/add/page"
```

File: tests/behat/features/content_types/article_view.feature (hunk 5 above)
Comment (omarlopesino) [inline tests/behat/features/content_types/article_view.feature]:
```suggestion
    Given I am an anonymous user
    When I go to the "node" entity with label "Behat article test"
```

File: tests/behat/features/content_types/article_delete.feature [hunk 15]
```diff
@@ -0,0 +1,37 @@
...
+    Then the response status code should be 403
+
+  @error @api @article @delete @authenticated
+  Scenario Outline: Simple Article delete authenticated access
+    When I go to "delete" of the "node" entity with label "Behat article test"
```
Comment (omarlopesino) [inline tests/behat/features/content_types/article_delete.feature]:
```suggestion
    Given I am logged in as a user with the <role> role
    When I go to "delete" of the "node" entity with label "Behat article test"
```

File: tests/behat/features/content_types/article_create.feature [hunk 16]
```diff
@@ -0,0 +1,42 @@
...
+    Then the response status code should be 403
+
+  @error @api @article @create @authenticated
+  Scenario Outline: Simple Article creation authenticated access
+    When I go to "/node/add/article"
```
Comment (omarlopesino) [inline tests/behat/features/content_types/article_create.feature]:
```suggestion
    Given I am logged in as a user with the <role> role
    When I go to "/node/add/article"
```

File: tests/behat/features/content_types/page_edit.feature [hunk 17]
```diff
@@ -0,0 +1,36 @@
...
+    Then the response status code should be 403
+
+  @error @api @page @edit @authenticated
+  Scenario Outline: Simple Page edit authenticated access
+    When I go to "edit" of the "node" entity with label "Behat page test"
```
Comment (omarlopesino) [inline tests/behat/features/content_types/page_edit.feature]:
```suggestion
    Given I am logged in as a user with the <role> role
    When I go to "edit" of the "node" entity with label "Behat page test"
```

**Review by omarlopesino — CHANGES_REQUESTED** [review]

**Review by omarlopesino — APPROVED** [review]

**Pull request merged**